client/components/tinymce/plugins/wpcom-view/views/embed/index.js

```diff
@@ -1,102 +1,60 @@
 /**
  * External dependencies
  */
-
 import EventEmitter from 'events';
-import { defer, uniqueId } from 'lodash';
+
 /**
  * Internal dependencies
  */
-import EmbedsListStore from 'lib/embeds/list-store';
-import EmbedsStore from 'lib/embeds/store';
-import actions from 'lib/embeds/actions';
 import EmbedView from './view';
+import getSiteEmbeds from 'state/selectors/get-site-embeds';
 import { getSelectedSiteId } from 'state/ui/selectors';
-import { SELECTED_SITE_SUBSCRIBE, SELECTED_SITE_UNSUBSCRIBE } from 'state/action-types';
+import { getReduxStore, reduxDispatch, reduxGetState } from 'lib/redux-bridge';
+import { requestEmbeds } from 'state/embeds/actions';
 
 export default class EmbedViewManager extends EventEmitter {
 	constructor() {
 		super();
 
-		this.sitesListener = this.updateSite.bind( this );
+		getReduxStore().then( store => {
+			store.subscribe(
+				this.createListener( store, getSelectedSiteId, this.onChange.bind( this ) )
+			);
+			store.subscribe(
+				this.createListener( store, this.getCurrentSiteEmbeds, this.onChange.bind( this ) )
+			);
+		} );
 	}
 
 	onChange() {
 		this.emit( 'change' );
-		this.fetchSiteEmbeds();
-	}
-
-	updateSite( selectedSiteId ) {
-		const siteId = selectedSiteId || getSelectedSiteId( this.store.getState() );
-
-		if ( ! this.hasOwnProperty( 'siteId' ) ) {
-			// First update (after adding initial listener) should trigger a
-			// fetch, but not emit a change event
-			this.siteId = siteId;
-			this.fetchSiteEmbeds();
-		} else if ( this.siteId !== siteId ) {
-			// Subsequent updates should neither emit a change nor trigger a
-			// fetch unless the site has changed
-			this.siteId = siteId;
-			this.onChange();
-		}
 	}
 
-	addListener( event, listener, store ) {
-		super.addListener( event, listener );
-		if ( 'change' === event && 1 === this.listeners( event ).length ) {
-			this.store = store;
-			this.selectedSiteSubscriberId = uniqueId();
-			store.dispatch( {
-				type: SELECTED_SITE_SUBSCRIBE,
-				listener: this.sitesListener,
-			} );
-			this.embedsListListener = EmbedsListStore.addListener( this.onChange.bind( this ) );
-			this.embedsListener = EmbedsStore.addListener( this.onChange.bind( this ) );
-			this.updateSite();
-		}
+	getCurrentSiteEmbeds( state ) {
+		return getSiteEmbeds( state, getSelectedSiteId( state ) );
 	}
 
-	removeListener( event, listener ) {
-		super.removeListener( event, listener );
+	createListener( store, selector, callback ) {
```

Comment:
`createListener` also doesn't need to be a method. It would be also nice if I could pass an array of selectors:
```js
createListener( store, [ getSelectedSiteId, getCurrentSiteEmbeds ], callback );
```
That's somewhat similar to how hooks handle dependency arrays.

Reply:
I think I'd prefer the explicitness of the current way it's written. I hope you wouldn't mind if I keep it as-is? 😉

Reply:
OK, but I still think it doesn't need to be a method 🙂